src/firefly/js/visualize/ui/FileUploadViewPanel.jsx

```diff
@@ -72,36 +57,39 @@ const TABLES_ONLY_SUPPORTED_TYPES=[
     FileAnalysisType.Table,
 ];
 
-
-const isTablesOnly= () => getAppOptions()?.uploadPanelLimit==='tablesOnly';
-
-
 const uploadOptions = 'uploadOptions';
 
-let currentAnalysisResult, currentReport, currentSummaryModel, currentDetailsModel;
 const FILE_UPLOAD_KEY= 'file-upload-key-';
 let keyCnt=0;
 
-export function FileUploadViewPanel({setSubmitText}) {
+export function FileUploadViewPanel({setSubmitText, acceptMoc}) {
+
+    const {groupKey}= useContext(FieldGroupCtx);
 
-    const {isLoading,statusKey} = useStoreConnector(() => getComponentState(panelKey, {isLoading:false,statusKey:''}));
+    const {isLoading,statusKey} = useStoreConnector(() => getComponentState(groupKey, {isLoading:false,statusKey:''}));
     const isWsUpdating          = useStoreConnector(() => isAccessWorkspace());
-    const uploadSrc             = useStoreConnector(() => getFieldVal(panelKey, uploadOptions));
-    const {message, analysisResult, report, summaryModel, detailsModel} = useStoreConnector((oldState) => getNextState(oldState));
+    const uploadSrc             = useStoreConnector(() => getFieldVal(groupKey, uploadOptions));
+
+
+    const SUMMARY_TBL_ID = groupKey; //FileUploadAnalysis
+    const DETAILS_TBL_ID = groupKey + '-Details';
+    const UNKNOWN_FORMAT = 'UNKNOWN';
+    const summaryUiId = SUMMARY_TBL_ID + '-UI';
+    const detailsUiId = DETAILS_TBL_ID + '-UI';
+
+    let {message, analysisResult, report, summaryModel, detailsModel, prevAnalysisResult} = useStoreConnector((oldState) =>
+        getNextState(oldState, groupKey));
 
     const [loadingMsg,setLoadingMsg]= useState(() => '');
     const [uploadKey,setUploadKey]= useState(() => FILE_UPLOAD_KEY+keyCnt);
 
     useEffect(() => {
-        setSubmitText(getLoadButtonText());
-    },[report,setSubmitText, summaryModel, detailsModel]);
 
-    useEffect(() => {
-        if (message || (analysisResult && analysisResult !== currentAnalysisResult)) {
-            if (currentAnalysisResult !== analysisResult) {
-                currentAnalysisResult = analysisResult;
+        if (message || (analysisResult !== prevAnalysisResult)) {
+            if (prevAnalysisResult !== analysisResult) {
+                prevAnalysisResult = analysisResult;
             }
-            dispatchComponentStateChange(panelKey, {isLoading: false});
+            dispatchComponentStateChange(groupKey, {isLoading: false});
         }
     });
```

Comment:
This block is a bit weird to me.  Why does `prevAnalysisResult` need to be  exposed outside of `getNextState`?  If just so we can compare that it was different then previous value, then we can just say..
```
    useEffect(() => {
        if (message || analysisResult) dispatchComponentStateChange(groupKey, {isLoading: false});
    }, [message, analysisResult]);
```
This way, you don't have to expose`prevAnalysisResult` from getNextState

Reply:
Ah yes, that is much simpler. I think I got caught up in how the global variables were being used before while trying to change them to local (and passing them around as needed). Missed this improvement. I'll change this.